### ISSUE TITLE
Add a new Lint rule to check exact one space before every line of documentation comments

### ISSUE THREAD

**Issue by syedmurtaza108** [dart-lang/linter#4879]
**Bad**

``` dart
///Line1
///Line2
///Line3
const variableFour = 4;
```

**Good**

``` dart
/// Line1
/// Line2
/// Line3
const variableFour = 4;
```

**Comment by srawlins** [comment]
I'm not sure how common this is, but I've seen / used each with a distinction:

* `// Line1` is for a textual comment.
* `//Line1` is for commented out code.

Your examples are doc comments, with triple slash, and I don't think I have any reason to write `///Line1`, so 👍 . Also I've never known a good reason to _commit_ commented out code, so I guess I would be fine linting against `//Line1` as well 👍 .

**Comment by bwilkerson** [comment]
I've actually committed commented out code before, typically when I want to record some thoughts for the future, things like
```dart
// Uncomment the line below when github.com/.../issue/1234 is fixed
//doSomethingCoolThatsCurrentlyBroken();
```

I wouldn't want to enable a lint that disallowed that case.

More importantly, this is a style-related lint, and we currently have a very high bar (in terms of value to users) for style related lints. I'm not sure that this meets that bar.
